movement/watch_faces/complication/higher_lower_game_face.c

```diff
@@ -0,0 +1,357 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 Chris Ellis
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+// Emulator only: need time() to seed the random number generator.
+#if __EMSCRIPTEN__
+#include <time.h>
+#endif
+
+#include <stdlib.h>
+#include <string.h>
+#include "higher_lower_game_face.h"
+#include "watch_private_display.h"
+
+#define TITLE_TEXT "Hi-Lo"
+#define GAME_BOARD_SIZE 6
+#define MAX_BOARDS 40
+#define GUESSES_PER_SCREEN 5
+#define WIN_SCORE MAX_BOARDS * GUESSES_PER_SCREEN
+#define STATUS_DISPLAY_START 0
+#define BOARD_SCORE_DISPLAY_START 2
+#define BOARD_DISPLAY_START 4
+#define BOARD_DISPLAY_END 9
+#define MIN_CARD_VALUE 2
+#define MAX_CARD_VALUE 14
+#define FLIP_BOARD_DIRECTION false
+
+typedef struct card_t {
+    uint8_t value;
+    bool revealed;
+} card_t;
+
+typedef enum {
+    A, B, C, D, E, F, G
+} segment_t;
+
+typedef enum {
+    HL_GUESS_EQUAL,
+    HL_GUESS_HIGHER,
+    HL_GUESS_LOWER
+} guess_t;
+
+typedef enum {
+    HL_GS_TITLE_SCREEN,
+    HL_GS_GUESSING,
+    HL_GS_WIN,
+    HL_GS_LOSE,
+    HL_GS_SHOW_SCORE,
+} game_state_t;
+
+static game_state_t game_state = HL_GS_TITLE_SCREEN;
+static card_t game_board[GAME_BOARD_SIZE] = {0};
+static uint8_t guess_position = 0;
+static uint8_t score = 0;
+static uint8_t completed_board_count = 0;
+
+static uint8_t generate_random_number(uint8_t num_values) {
+    // Emulator: use rand. Hardware: use arc4random.

```

Comment:
Why the difference?

Reply:
I largely cribbed what some other faces did, but my understanding that is the emulator doesn't support the `arc4random` function, which I believe less predictable RNG.